src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs

```diff
@@ -15,6 +15,11 @@ namespace Microsoft.PowerFx
     /// </summary>
     public class ParserOptions
     {
+        /// <summary>
+        /// If true, This would allow users to do partial type checking by enabling support for Unknown type in input.
+        /// </summary>
+        public bool AllowDeferredType { get; set; }
```

Comment:
Why does this need to be in ParserOptions? The parser shouldn't care about type checking features.